Design/Mantid_As_a _SciPy_Styled_Package.md

```diff
@@ -0,0 +1,259 @@
+Motivation
+==========
+
+Shaping Mantid as a SciPy styled package is one of the strategic aims in the [ORNL Mantid 5 year plan](ORNL_Mantid_5yearplan.pdf).
+However it has become clear over several meetings and workshops that what should be included as part of this aim is not commonly agreed or 
+understood within the development team or management of the project.
+This document is an attempt to harmonise this understanding or at least provide a center point for discussion and agreement or what we want 
+to achieve as part of this strategic aim and to provide a baseline high level design of what we need to do to implement it.
+
+Requirements
+============
+
+As mentioned above the requirements underlying the strategic aim are not universally shared so this document will serve as a place to detail
+what we have agreed.
+
+This is an initial draft from a discussion between Nick Draper and Martyn Gigg and will need to be discussed and agreed with 
+other team leads.
+
+Must
+----
+
+1. The Algorithms need to be separated into sensible sub packages that allow sensible imports of only the areas of interest 
+   rather than all algorithms as is the current situation.
+1. The current top level api import must be maintained in name and function. 
+   So `from mantid.simpleapi import *` must import all of the sub packages.
+1. The documentation structure of Algroithms must match that of the sub packages to make it easier to locate the documents you need.
+1. The category structure in Mantid must match the python package structure, but use english language syntax rather than Python syntax.
+   For example Neutron vs Mantid.Neutron.
+1. We must maintain backward compatability with the majority of users scripts, small changes are permissible if we are confident that this will not affect a significant number of users or scripts.  All breaking changes must be notified in advance of the release, ideally supporting deprecated aspects for at least one release.
+1. All documentation and training courses will need to be updated to match the new structures.
+
+   
+
+
+Should
+------
+
+1. The structure of the python sub packages and other structures should be based on that presented by Andre Savici at the 
+   [2017 Users Workshop](../Presentations/DevMeetings/2017-06/Mantid4PythonLibrariesAndNames.md).  
+   Note: this will not include the changes to the naming of algorithms, just the package structuring.
+1. We should take advantage of the restructuring into smaller packages to define a small number of these to be "core".
+   This will define this term, which currently is interpreted by each individual.
+1. The C++ library structure should mirror the python package structure as far as possible to make it easy for developers to 
+   navigate the code.
+1. We should decrease the time it takes to generate the Python API at run time. 
```

Comment:
This is a non-functional/operational requirement. Also, I would say that it more important that a "Should"